source/mir/bignum/low_level_view.d

```diff
@@ -122,6 +122,211 @@ version(mir_bignum_test)
     assert(ceilLog10Exp2(10UL) == 4); // ulong
 }
 
+/++
+Performs `q = u / v, r = u % v` (unsigned)
+Preconditions:
+    * non-empty coefficients
+    * little-endian order
+    * u.length == m+1
+    * v.length == n
+    * q.length == (m - n) + 1 (IF m < n, this is not needed)
+    * r.length == n
+NOTE: This function mutates the input arguments. Beware.
++/
+void divm(W, WordEndian endian = TargetEndian)(BigUIntView!(W, endian) _u,
+                                               BigUIntView!(W, endian) _v,
+                                               BigUIntView!(W, endian) _q,
+                                               BigUIntView!(W, endian) _r,
+                                               size_t m, size_t n)
+    @trusted pure nothrow @nogc
+if (is(Unqual!W == uint))
+{
+    // Adopted from Hacker's Delight (2nd Edition), 4-2
+    // License permits inclusion:
+    // You are free to use, copy, and distribute any of the code on this web site, whether modified by you or not.
+    // You need not give attribution. This includes the algorithms (some of which appear in Hacker's Delight), the Hacker's Assistant, and any code submitted by readers. Submitters implicitly agree to this.
+    // The textural material and pictures are copyright by the author, and the usual copyright rules apply. E.g., you may store the material on your computer and make hard or soft copies for your own use.
+    // However, you may not incorporate this material into another publication without written permission from the author (which the author may give by email).
+    // The author has taken care in the preparation of this material, but makes no expressed or implied warranty of any kind and assumes no responsibility for errors or omissions.
+    // No liability is assumed for incidental or consequential damages in connection with or arising out of the use of the information or programs contained herein. 
+    import mir.bitop: ctlz;
+    import mir.utility: _expect;
+
+    enum shift = (W.sizeof * 8);
+    enum b = ulong(1) << shift;
+
+    // Convert all of our input arguments to little-endian
+    auto u = _u.leastSignificantFirst();
+    auto v = _v.leastSignificantFirst();
+    auto q = _q.leastSignificantFirst();
+    auto r = _r.leastSignificantFirst();
+
+    assert(m != 0 && n != 0);
+    assert(_u.coefficients.length >= m + 1, "Dividend array is too small (should be m+1)");
+    assert(_v.coefficients.length >= n, "Divisor array is too small");
+    assert(_r.coefficients.length >= n, "Remainder array is too small");
+    assert(v[n - 1] != 0);
+
+    // Handle a common edge case with division (where the dividend is smaller then the divisor)
+    // We don't care about how big the quotient array is here (and in fact, our math will fail)
+    // since we just dump the dividend into the remainder 
+    if (_expect(m < n, false))
+    {
+        foreach(i; 0 .. m)
```

Comment:
`r[0 .. m] = u;`

Reply:
This one is unneeded, as I was able to entirely phase out the `r` param